### PR TITLE
Adding a login api in the registration app

### PR DESCRIPTION
**Pull request by ghost**
**Issue Link**
Asana: [Link](https://app.asana.com/0/1202718495875213/1202725278795575/f)

**Definition of Done**
- Users can log in to their accounts by inputting their username and password
- Users can only log in if they have registered their account.

**Notes:**

- Open Link: http://localhost:8000/api/user/login/
Logging in using registered user
Logging in using unregistered user

### CODE REVIEW

File: backend/registration/urls.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
-from .views import RegisterAPI
+from .views import RegisterAPI,LoginAPI
 from django.urls import path
 
 urlpatterns = [
     path('api/user/register', RegisterAPI.as_view(), name='register'),
+    path('api/user/login/', LoginAPI.as_view(), name='login'),
```
Comment (kentlouise) [inline backend/registration/urls.py]: Update to `path('api/user/login', LoginAPI.as_view(), name='login'),`

File: backend/registration/urls.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
-from .views import RegisterAPI
+from .views import RegisterAPI,LoginAPI
```
Comment (kentlouise) [inline backend/registration/urls.py]: Add extra space.
`from .views import RegisterAPI, LoginAPI`

**Review by chevyB — CHANGES_REQUESTED** [review]
I wonder why you'd change the implementation.

This authentication scheme is only applicable if you are using django as your frontend. This will not work if you have another frontend technology since it needs csrf token which can only be generated when you are at the same `session` of your backend server (django). While if you are in react, there is no way to generate that one.

So best thing to use is knox.

File: backend/registration/urls.py (hunk 1 above)
Comment (kentlouise) [reply backend/registration/urls.py]: @aika-blanes 
I think its a best practice not to include a forward slash at the end of the api endpoint.

Maybe we can update this to:
`'api/user/login'`

References:
- https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/
- https://www.freecodecamp.org/news/rest-api-best-practices-rest-endpoint-design-examples/
- https://stackoverflow.com/questions/7140074/restfully-design-login-or-register-resources

**Review by kentlouise — COMMENTED** [review]
One last comment. Your implementation is already working fine. However this requires me to have a django-filter. Can you add the package `django-filter` in `backend/requirements.txt`.

```python
django-cors-headers
djangorestframework
django-summernote
django-rest-knox
django-seed
django-filter
```

**Review by kentlouise — APPROVED** [review]
# Approved #15 
- Code looks good to me.
- Implementation works well.

[test.webm](https://user-images.githubusercontent.com/87509953/184789977-4b693654-e33a-42d8-836e-27c1d66de340.webm)

**Review by chevyB — APPROVED** [review]

**Pull request merged**